src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -116,6 +116,14 @@ namespace AppInstaller::CLI::Workflow
             context.Reporter.Error() << Resource::String::NoApplicableInstallers << std::endl;
             AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_NO_APPLICABLE_INSTALLER);
         }
+
+        // This installer cannot be run elevated, but we are running elevated.
+        // Implementation of de-elevation is complex; simply block for now.
+        if (installer->ElevationRequirement == ElevationRequirementEnum::ElevationProhibited && Runtime::IsRunningAsAdmin())

```

Comment:
>if (installer->ElevationRequirement == ElevationRequirementEnum::ElevationProhibited && Runtime::IsRunningAsAdmin()) [](http://example.com/codeflow?start=8&length=116)

I'm sort of mixed. Did you think of moving to installer selection in case there're other applicable installers? Or it's preferred to tell the user to run non-elevated again since this is what we think is the best installer.

Reply:
I think it may be best to have it in the installer, especially given that there is a [feature request](https://github.com/microsoft/winget-cli/issues/1898) to filter by elevation requirement. However, if there are no applicable installers based solely on the fact that the user is running in an elevated terminal, then it still should say elevation prohibited rather than no applicable installer found

Reply:
I don't think that installer selection is appropriate, we don't even have user intent without #1898.  I don't consider "is winget running as admin or not" user intent here, because I want to be able to run my Terminal as user and elevate installers as needed.  #1898 should be implemented in installer selection (as well as search) though, because we would have user intent.